documentation/dotnet-monitor.md

```diff
@@ -0,0 +1,276 @@
+
+## Tour of dotnet-monitor
+
+### Installation
+
+To get started with `dotnet-monitor` locally, you will need to have [.NET Core](https://dotnet.microsoft.com/download) installed on your machine. `dotnet-monitor` can then be installed as a global tool using the following command:
+
+```bash
+dotnet tool install -g dotnet-monitor --add-source https://dnceng.pkgs.visualstudio.com/public/_packaging/dotnet5-transport/nuget/v3/index.json --version 5.0.0-preview*
+```
+
+When consuming `dotnet-monitor` as a container image, it can be pulled from [MCR](https://hub.docker.com/_/microsoft-dotnet-nightly-monitor/) using the following command:
+
+```bash
+docker pull mcr.microsoft.com/dotnet/nightly/monitor:5.0.0-preview1
+```
+
+### Running `dotnet-monitor`
+
+Once installed you can run `dotnet-monitor` via the following command,
+
+```bash
+dotnet monitor collect
+```
+
+When running in a docker container, you will need to share a volume mount between your application container and `dotnet-monitor`
+
+```bash
+docker volume create diagnosticserver
+docker run -d --rm -p 8000:80 -v diagnosticsserver:/tmp mcr.microsoft.com/dotnet/core/samples:aspnetapp
+docker run -it --rm -p 52323:52323 -v diagnosticsserver:/tmp mcr.microsoft.com/dotnet/nightly/monitor --urls http://*:52323
+```
+
+When running in cluster, it is recommend to run the `dotnet-monitor` container as a sidecar alongside your application container in the pod. This sample Kubernetes manifest shows how to configure your deployment to include a sidecar container.
+
+```yaml
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: dotnet-hello-world
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      app: dotnet-hello-world
+  template:
+    metadata:
+      labels:
+        app: dotnet-hello-world
+    spec:
+      volumes:
+      - name: diagnostics
+        emptyDir: {}
+      containers:
+      - name: server
+        image: mcr.microsoft.com/dotnet/core/samples:aspnetapp
+        ports:
+        - containerPort: 80
+        volumeMounts:
+          - mountPath: /tmp
+            name: diagnostics
+      - name: sidecar
+        image: mcr.microsoft.com/dotnet/nightly/monitor
+        ports:
+        - containerPort: 52325
+        # args: ["--urls", "http://*:52323", "--metricUrls", "http://*:52325"]
+        volumeMounts:
+          - name: diagnostics
+            mountPath: /tmp
+```
+
+
+
+In the default configuration `dotnet-monitor` binds to two different  groups of URLs. The URLs controlled via the `--urls` parameter (defaults to http://localhost:52323) expose all the collection endpoints. The URLs controlled via the ``--metricUrls` parameter (defaults to http://localhost:52325) only exposes the `/metrics` endpoint. Since diagnostics artifacts such as logs, dumps, and traces can leak sensitive information about the application, it is **strongly recommended** that you only expose the metrics URL in the container scenario.
```

Comment:
I guess expose is an overloaded term. It could mean make available to other services in your cluster or it could mean make available to the public internet.

In this case, I meant make available to other services in your cluster. I'll be explicit

Reply:
I changed the guidance up front to recommend not to expose any endpoint. I will go into greater details in the metrics section.